python/templates/macros/collections.jinja2

```diff
@@ -71,9 +71,19 @@ std::vector<{{ member.full_type }}> {{ class.bare_type }}Collection::{{ member.n
         m_rel_{{ relation.name }}->emplace_back({{ relation.full_type }}::makeEmpty());
         continue;
       }
+{% if relation.interface_types %}
+      // We need the concrete collection type to assign it to an InferenceWrapper
+{% for int_type in relation.interface_types %}
+      if (auto concreteColl = dynamic_cast<{{ int_type.full_type }}Collection*>(coll)) {
```

Comment:
This should result in an `if`-`else if` chain, rather than just a list of `if`s.